### PR TITLE
Add test to verify RemoveImports works inside active #if clauses

### PR DESCRIPTION
**Pull request by amanthatdoescares**
### This PR expands test coverage for the Remove Unused Imports code action.
Fix: https://github.com/swiftlang/sourcekit-lsp/issues/2335
#### Existing tests verify that:
Unused top-level import declarations are removed correctly
The code action is not available when the source file contains compilation errors

This PR begins adding test coverage for import declarations nested inside conditional compilation blocks. In particular, it verifies that unused imports inside active #if clauses are considered by the code action.

This helps guard against regressions when extending RemoveUnusedImports to account for conditional compilation regions in future changes.

### CODE REVIEW

File: Tests/SourceKitLSPTests/CodeActionTests.swift [hunk 1]
```diff
@@ -1649,6 +1143,118 @@ final class CodeActionTests: SourceKitLSPTestCase {
     )
   }
 
+  func testRemoveUnusedImportsFromActiveIfClause() async throws {
+    let project = try await SwiftPMTestProject(
+      files: [
+        "Test/Test.swift": """
+        1️⃣#if canImport(Darwin)
```
Comment (rintaro) [inline Tests/SourceKitLSPTests/CodeActionTests.swift]: `canImport(Darwin)` is not active on non-Darwin environment. Can we use more reliable tests e.g.
In `Package.swift`
```swift
swiftSettings: [.define("FLAG")]
```
then in `Test/Test.swift`
```
#if FLAG
```

File: Tests/SourceKitLSPTests/CodeActionTests.swift [hunk 2]
```diff
@@ -1649,6 +1143,118 @@ final class CodeActionTests: SourceKitLSPTestCase {
...
+        arguments: removeUnusedImportsCommand.arguments
+      )
+    )
+  }
+
+  func testRemoveUnusedImportsDoesNotRemoveImportsFromInactiveIfClause() async throws {
+    let project = try await SwiftPMTestProject(
+      files: [
+        "Test/Test.swift": """
+        1️⃣#if canImport(Darwin)
```
Comment (rintaro) [inline Tests/SourceKitLSPTests/CodeActionTests.swift]: Ditto. Please avoid using `canImport`

File: Tests/SourceKitLSPTests/CodeActionTests.swift (hunk 1 above)
Comment (rintaro) [reply Tests/SourceKitLSPTests/CodeActionTests.swift]: Also, do you know why this is rendered as "1□" but the `CodeActionResult` call below is displayed as "1️⃣"  in my GitHub UI? 🤔 The UTF8 code sequence is different?

Image

Comment (amanmaurya92) [reply Tests/SourceKitLSPTests/CodeActionTests.swift]: image
The marker is the keycap emoji (1️⃣), which is composed of multiple Unicode code points. GitHub’s diff viewer likely doesn’t fully render the combining keycap, so it shows up as 1□. The UTF-8 sequence is indeed different, but this is expected.

File: Tests/SourceKitLSPTests/CodeActionTests.swift (hunk 2 above)
Comment (amanmaurya92) [reply Tests/SourceKitLSPTests/CodeActionTests.swift]: Thanks for the suggestion — I’ve replaced canImport(...) with a defined flag to make the test deterministic across environments. Please let me know if there are additional scenarios or edge cases you’d like to see covered in this PR to make the tests more robust.

**Comment by amanmaurya92** [comment]
@rintaro I’m also interested in working on improving the RemoveUnusedImports code action itself, and thought it would be useful to add these tests first to lock in the expected behaviour around conditional compilation before making any implementation changes.

**Comment by rintaro** [comment]
@swift-ci Please test

File: Tests/SourceKitLSPTests/CodeActionTests.swift [hunk 3]
```diff
@@ -1649,6 +1143,125 @@ final class CodeActionTests: SourceKitLSPTestCase {
...
+      )
+    )
+  }
+
+  func testRemoveUnusedImportsDoesNotRemoveImportsFromInactiveIfClause() async throws {
+    let project = try await SwiftPMTestProject(
+      files: [
+        "Test/Test.swift": """
+        1️⃣#if FLAG
+        import Darwin
```
Comment (rintaro) [inline Tests/SourceKitLSPTests/CodeActionTests.swift]: I guess this still fails in non-Darwin environments because `import Darwin` causes compilation errors.

Comment (amanmaurya92) [reply Tests/SourceKitLSPTests/CodeActionTests.swift]: yes, i changed it to Foundation now.

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test Windows

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@amanthatdoescares The tests you added don’t pass because they test functionality that doesn’t exist yet. Please ensure that the test you add pass locally before putting them up for PR review.

**Comment by ahoppen** [comment]
@DPrakashhh started fixing this issue in https://github.com/swiftlang/sourcekit-lsp/pull/2457. The comments in that PR would also apply here. Also, I don’t want to have two concurrent PRs implementing the same feature and since https://github.com/swiftlang/sourcekit-lsp/pull/2457 already has already gone through some iterations, I’d prefer that.
